internal/settings/providers/namecom/createrecord.go

```diff
@@ -0,0 +1,66 @@
+package namecom
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/qdm12/ddns-updater/internal/settings/constants"
+	"github.com/qdm12/ddns-updater/internal/settings/errors"
+	"github.com/qdm12/ddns-updater/internal/settings/utils"
+)
+
+func (p *Provider) createRecord(ctx context.Context, client *http.Client,
+	ip net.IP) (err error) {
+	recordType := constants.A
+	if ip.To4() == nil {
+		recordType = constants.AAAA
+	}
+
+	u := &url.URL{
+		Scheme: "https",
+		Host:   "api.name.com",
+		Path:   fmt.Sprintf("/v4/domains/%s/records", p.domain),
+		User:   url.UserPassword(p.username, p.password),
+	}
+
+	postRecordsParams := struct {
+		Host   string `json:"host"`
+		Type   string `json:"type"`
+		Answer string `json:"answer"`
+		TTL    int    `json:"ttl"`
+	}{
+		Host:   p.host,
+		Type:   recordType,
+		Answer: ip.String(),
+		TTL:    p.ttl,
+	}
+
+	bodyBytes, err := json.Marshal(postRecordsParams)
+	if err != nil {
+		return fmt.Errorf("%w: %w", errors.ErrRequestMarshal, err)
+	}
+
+	request, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), bytes.NewBuffer(bodyBytes))
+	if err != nil {
+		return fmt.Errorf("%w: %w", errors.ErrBadRequest, err)
+	}
+
+	response, err := client.Do(request)
+	if err != nil {
+		return fmt.Errorf("doing HTTP request: %w", err)
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		return fmt.Errorf("%w: %d: %s",
+			errors.ErrBadHTTPStatus, response.StatusCode,
+			utils.BodyToSingleLine(response.Body))
```

Comment:
Let's parse the error message

> Any non-200 response will return an ErrorMessage JSON response. ErrorMessages have a "message" that gives a general error message st ring, and an optional "details" key which contains a string with additional information about the error. 

Common API errors are listed at https://www.name.com/api-docs and you should match them to existing errors (from `internal/settings/errors`)